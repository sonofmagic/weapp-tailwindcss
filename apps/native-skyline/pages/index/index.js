import { cityData } from './data'
import { add } from 'lodash'
Component({
  data: {
    list: [],
  },
  lifetimes: {
    attached() {
      const cities = cityData
      // 按拼音排序
      cities.sort((c1, c2) => {
        const pinyin1 = c1.pinyin.join('')
        const pinyin2 = c2.pinyin.join('')
        return pinyin1.localeCompare(pinyin2)
      })
      // 添加首字母
      const map = new Map()
      for (const city of cities) {
        const alpha = city.pinyin[0].charAt(0).toUpperCase()
        if (!map.has(alpha)) map.set(alpha, [])
        map.get(alpha).push({ name: city.fullname })
      }

      const keys = []
      for (const key of map.keys()) {
        keys.push(key)
      }
      keys.sort()

      const list = []
      for (const key of keys) {
        list.push({
          alpha: key,
          subItems: map.get(key)
        })
      }

      console.log('address-book list:', list)
      this.setData({ list })
      console.log(add(1,3));
    },
  },
})
