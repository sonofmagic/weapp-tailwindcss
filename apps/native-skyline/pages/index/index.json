{
  "usingComponents": {
    "navigation-bar": "../../components/navigation-bar",
    "address-book": "../../components/address-book"
  },
  "disableScroll": true,
  "navigationStyle": "custom"
}