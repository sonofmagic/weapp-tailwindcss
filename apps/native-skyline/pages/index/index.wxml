<navigation-bar title="通讯录" back="{{true}}"></navigation-bar>
<view class="w-screen grow overflow-hidden">
  <address-book list="{{list}}">
    <view class="text-black/90">
      <view class="p-10">
        <view class="page__title">Address Book</view>
        <view class="page__desc">类通讯录列表</view>
        <view class="i-mdi-home bg-yellow-600 w-4 h-4"></view>
      </view>
      <view class="pb-10">
      </view>
    </view>
  </address-book>
</view>
