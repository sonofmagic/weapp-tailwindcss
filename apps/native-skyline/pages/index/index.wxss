page {
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  height: 100vh;
  background-color: #f7f7f7;
}

.page__title {
  @apply text-center text-xl;
}

.page__desc {
  @apply text-center text-sm;
  margin-top: 5px;
  color: rgba(0, 0, 0, 0.5);
}

.cell {
  height: 50px;
  justify-content: center;
  align-items: center;
}
