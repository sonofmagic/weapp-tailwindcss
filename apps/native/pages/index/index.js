// index.js
// 获取应用实例
const app = getApp()

Page({
  data: {
    motto: 'Hello World',
    userInfo: {},
    hasUserInfo: false,
    canIUse: wx.canIUse('button.open-type.getUserInfo'),
    imgUrls: null,
    indicatorDots: true,
    autoplay: true,
    interval: 5000,
    duration: 1000,
    list: [],
    page: 1
  },
  // 事件处理函数
  bindViewTap: function () {
    wx.navigateTo({
      url: '../logs/logs'
    })
  },
  addCart(data) {
    const item = data.currentTarget.dataset.item
  },
  imgsc: function (e) {
    wx.chooseImage({
      count: 1, // 默认9
      sizeType: ['original', 'compressed'], // 可以指定是原图还是压缩图，默认二者都有
      sourceType: ['album', 'camera'], // 可以指定来源是相册还是相机，默认二者都有
      success: function (res) {
        // 返回选定照片的本地文件路径列表，tempFilePath可以作为img标签的src属性显示图片
        const tempFilePaths = res.tempFilePaths
      }
    })
  },
  lower: function (e) {
    console.log(e)
    this.getList()
  },
  getList: function () {},
  onLoad: function () {
    const app = getApp()
    this.getList()

    if (app.globalData.userInfo) {
      this.setData({
        userInfo: app.globalData.userInfo,
        hasUserInfo: true
      })
    } else if (this.data.canIUse) {
      // 由于 getUserInfo 是网络请求，可能会在 Page.onLoad 之后才返回
      // 所以此处加入 callback 以防止这种情况
      app.userInfoReadyCallback = (res) => {
        this.setData({
          userInfo: res.userInfo,
          hasUserInfo: true
        })
      }
    } else {
      // 在没有 open-type=getUserInfo 版本的兼容处理
      wx.getUserInfo({
        success: (res) => {
          app.globalData.userInfo = res.userInfo
          this.setData({
            userInfo: res.userInfo,
            hasUserInfo: true
          })
        }
      })
    }
  },
  changeIndicatorDots: function (e) {
    this.setData({
      indicatorDots: !this.data.indicatorDots
    })
  },
  changeAutoplay: function (e) {
    this.setData({
      autoplay: !this.data.autoplay
    })
  },
  intervalChange: function (e) {
    this.setData({
      interval: e.detail.value
    })
  },
  durationChange: function (e) {
    this.setData({
      duration: e.detail.value
    })
  },
  getUserInfo: function (e) {
    app.globalData.userInfo = e.detail.userInfo
    this.setData({
      userInfo: e.detail.userInfo,
      hasUserInfo: true
    })
  }
})
