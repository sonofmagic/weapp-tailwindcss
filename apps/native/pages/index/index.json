{
  "navigationBarTitleText": "首页",
  "enablePullDownRefresh":true,
  "backgroundTextStyle":"dark"
}