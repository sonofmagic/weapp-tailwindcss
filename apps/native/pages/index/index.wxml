<!-- index.wxml -->
<scroll-view class="scrollarea" scroll-y type="list">
  <view class="container">
    <view class="userinfo">
      <block wx:if="{{canIUseNicknameComp && !hasUserInfo}}">
        <button open-type="chooseAvatar" bind:chooseavatar="onChooseAvatar">
          <image class="w-[100px] h-[100px]" src="{{userInfo.avatarUrl}}"></image>
        </button>
        <view class="nickname-wrapper">
          <text class="nickname-label">昵称</text>
          <input type="nickname" class="nickname-input" placeholder="请输入昵称" bind:change="onInputChange" />
        </view>
        <view class="text-[#432432]">{{ num }}</view>
      </block>
      <block wx:elif="{{!hasUserInfo}}">
        <button wx:if="{{canIUseGetUserProfile}}" bindtap="getUserProfile">获取头像昵称</button>
        <view wx:else>请使用2.10.4及以上版本基础库</view>
      </block>
      <block wx:else>
        <image bindtap="bindViewTap" class="userinfo-avatar" src="{{userInfo.avatarUrl}}" mode="cover"></image>
        <text>{{userInfo.nickName}}</text>
      </block>
    </view>
    <view class="mt-[40px]">
      <text class="{{ motto }}">{{motto}}</text>
    </view>
  </view>
</scroll-view>