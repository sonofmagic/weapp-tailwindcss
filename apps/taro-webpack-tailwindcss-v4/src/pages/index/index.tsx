import { Text, View } from '@tarojs/components'
import { useLoad } from '@tarojs/taro'
import './index.css'

export default function Index() {
  useLoad(() => {
    console.log('Page loaded.')
  })

  return (
    <>
      <View className="bg-[#2e2bcc] text-[100rpx] text-white">
        <Text>Hello world!</Text>
      </View>
      <View className="bg-purple-800 text-pink-200">
        11
      </View>
      <View className="h-14 bg-gradient-to-r from-cyan-500 to-blue-500"></View>
      <View className="bg-purple-800 text-pink-200 invert">
        invert
      </View>
      <View className="bg-purple-800 text-pink-200 backdrop-blur-2xl">
        backdrop-blur-2xl
      </View>
      <View className="skew-x-0 bg-purple-800 text-pink-200">
        skew-x-0
      </View>
      <View className="flex space-x-1 space-y-1.5">
        <View className="bg-purple-800 text-pink-200">
          Tailwind
        </View>
        <View className="bg-purple-800 text-pink-200">
          Tailwind
        </View>
        <View className="bg-purple-800 text-pink-200">
          Tailwind
        </View>
        <Text className="bg-purple-800 text-pink-200">
          Tailwind
        </Text>
      </View>
      <View className="flex flex-col space-y-1.5">
        <View className="bg-purple-800 text-pink-200">
          Tailwind CSS
        </View>
        <View className="bg-purple-800 text-pink-200">
          Tailwind CSS
        </View>
        <View className="bg-purple-800 text-pink-200">
          Tailwind CSS
        </View>
        <Text className="bg-purple-800 text-pink-200">
          Tailwind CSS
        </Text>
      </View>
    </>

  )
}
