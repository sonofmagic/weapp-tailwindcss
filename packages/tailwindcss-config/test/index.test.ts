import path from 'pathe'
import { loadConfig } from '@/index'

const fixturesDir = path.resolve(import.meta.dirname, './fixtures')

vi.mock(import('@/index'), async (importOriginal) => {
  const { loadConfig: _loadConfig } = await importOriginal()

  function loadConfig(...args: Parameters<typeof _loadConfig>) {
    return _loadConfig(...args).then((x) => {
      if (x && 'filepath' in x) {
        // @ts-ignore
        delete x.filepath
      }
      return x
    })
  }

  return {
    loadConfig,
  }
})

describe('index', () => {
  it('cwd', async () => {
    expect(await loadConfig({
      cwd: path.resolve(fixturesDir, './cwd'),
    })).toMatchSnapshot()
  })

  it('cjs', async () => {
    expect(await loadConfig({
      cwd: path.resolve(fixturesDir, './cjs'),
    })).toMatchSnapshot()
  })

  it('js', async () => {
    expect(await loadConfig({
      cwd: path.resolve(fixturesDir, './js'),
    })).toMatchSnapshot()
  })

  it('ts', async () => {
    expect(await loadConfig({
      cwd: path.resolve(fixturesDir, './ts'),
    })).toMatchSnapshot()
  })

  it('mjs', async () => {
    expect(await loadConfig({
      cwd: path.resolve(fixturesDir, './mjs'),
    })).toMatchSnapshot()
  })
})
