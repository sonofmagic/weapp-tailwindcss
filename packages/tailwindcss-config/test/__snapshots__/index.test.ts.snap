// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`index > cjs 1`] = `
{
  "config": {
    "content": [
      "./src/**/*.{html,js}",
    ],
    "theme": {
      "colors": {
        "blue": "#1fb6ff",
        "gray": "#8492a6",
        "gray-dark": "#273444",
        "gray-light": "#d3dce6",
        "green": "#13ce66",
        "orange": "#ff7849",
        "pink": "#ff49db",
        "purple": "#7e5bef",
        "yellow": "#ffc82c",
      },
      "extend": {
        "borderRadius": {
          "4xl": "2rem",
        },
        "spacing": {
          "8xl": "96rem",
          "9xl": "128rem",
        },
      },
      "fontFamily": {
        "sans": [
          "Graphik",
          "sans-serif",
        ],
        "serif": [
          "Merriweather",
          "serif",
        ],
      },
    },
  },
}
`;

exports[`index > config option resolves explicit file path 1`] = `
{
  "config": {
    "default": {
      "content": [
        "./src/**/*.{html,js}",
      ],
      "theme": {
        "colors": {
          "blue": "#1fb6ff",
          "gray": "#8492a6",
          "gray-dark": "#273444",
          "gray-light": "#d3dce6",
          "green": "#13ce66",
          "orange": "#ff7849",
          "pink": "#ff49db",
          "purple": "#7e5bef",
          "yellow": "#ffc82c",
        },
        "extend": {
          "borderRadius": {
            "4xl": "2rem",
          },
          "spacing": {
            "8xl": "96rem",
            "9xl": "128rem",
          },
        },
        "fontFamily": {
          "sans": [
            "Graphik",
            "sans-serif",
          ],
          "serif": [
            "Merriweather",
            "serif",
          ],
        },
      },
    },
  },
}
`;

exports[`index > cwd 1`] = `
{
  "config": {
    "content": [
      "./src/**/*.{html,js}",
    ],
    "theme": {
      "colors": {
        "blue": "#1fb6ff",
        "gray": "#8492a6",
        "gray-dark": "#273444",
        "gray-light": "#d3dce6",
        "green": "#13ce66",
        "orange": "#ff7849",
        "pink": "#ff49db",
        "purple": "#7e5bef",
        "yellow": "#ffc82c",
      },
      "extend": {
        "borderRadius": {
          "4xl": "2rem",
        },
        "spacing": {
          "8xl": "96rem",
          "9xl": "128rem",
        },
      },
      "fontFamily": {
        "sans": [
          "Graphik",
          "sans-serif",
        ],
        "serif": [
          "Merriweather",
          "serif",
        ],
      },
    },
  },
}
`;

exports[`index > js 1`] = `
{
  "config": {
    "default": {
      "content": [
        "./src/**/*.{html,js}",
      ],
      "theme": {
        "colors": {
          "blue": "#1fb6ff",
          "gray": "#8492a6",
          "gray-dark": "#273444",
          "gray-light": "#d3dce6",
          "green": "#13ce66",
          "orange": "#ff7849",
          "pink": "#ff49db",
          "purple": "#7e5bef",
          "yellow": "#ffc82c",
        },
        "extend": {
          "borderRadius": {
            "4xl": "2rem",
          },
          "spacing": {
            "8xl": "96rem",
            "9xl": "128rem",
          },
        },
        "fontFamily": {
          "sans": [
            "Graphik",
            "sans-serif",
          ],
          "serif": [
            "Merriweather",
            "serif",
          ],
        },
      },
    },
  },
}
`;

exports[`index > mjs 1`] = `
{
  "config": {
    "default": {
      "content": [
        "./src/**/*.{html,js}",
      ],
      "theme": {
        "colors": {
          "blue": "#1fb6ff",
          "gray": "#8492a6",
          "gray-dark": "#273444",
          "gray-light": "#d3dce6",
          "green": "#13ce66",
          "orange": "#ff7849",
          "pink": "#ff49db",
          "purple": "#7e5bef",
          "yellow": "#ffc82c",
        },
        "extend": {
          "borderRadius": {
            "4xl": "2rem",
          },
          "spacing": {
            "8xl": "96rem",
            "9xl": "128rem",
          },
        },
        "fontFamily": {
          "sans": [
            "Graphik",
            "sans-serif",
          ],
          "serif": [
            "Merriweather",
            "serif",
          ],
        },
      },
    },
  },
}
`;

exports[`index > ts 1`] = `
{
  "config": {
    "default": {
      "content": [
        "./src/**/*.{html,js}",
      ],
      "theme": {
        "colors": {
          "blue": "#1fb6ff",
          "gray": "#8492a6",
          "gray-dark": "#273444",
          "gray-light": "#d3dce6",
          "green": "#13ce66",
          "orange": "#ff7849",
          "pink": "#ff49db",
          "purple": "#7e5bef",
          "yellow": "#ffc82c",
        },
        "extend": {
          "borderRadius": {
            "4xl": "2rem",
          },
          "spacing": {
            "8xl": "96rem",
            "9xl": "128rem",
          },
        },
        "fontFamily": {
          "sans": [
            "Graphik",
            "sans-serif",
          ],
          "serif": [
            "Merriweather",
            "serif",
          ],
        },
      },
    },
  },
}
`;
